Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -330,197 +845,125 @@ sample_triangle_mesh(const TriangleMesh& tm,
-          nb_points = num_vertices(tm);
-        else
-          nb_points = static_cast<std::size_t>(
-            std::ceil( g.mesh_length()*nb_pts_a_u) );
-      }
-      out = std::copy_n(g, nb_points, out);
-    }
-  }
+/** \ingroup PMP_distance_grp
+ * generates points taken on `triangles` and outputs them to `out`, the sampling method
+ * is selected using named parameters.
+ * @tparam PointRange  a model of the concept `RandomAccessContainer` whose value type is the point type.
+ * @tparam TriangleRange a model of the concept `RandomAccessContainer`
+ *                      whose value_type is itself a model of the concept `RandomAccessContainer`
+ *                      whose value_type is `std::size_t`.
+ * @tparam OutputIterator a model of `OutputIterator`
+ *  holding point type objects.
+ *
+ * @param points the points of the soup that will be sampled.
+ * @param triangles a vector containing the triangles of the soup that will be sampled.
+ * @param out output iterator to be filled with sampled points
+ * @param np an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{geom_traits} a model of `PMPDistanceTraits`. \cgalParamEnd
+ *    \cgalParamBegin{use_random_uniform_sampling}
+ *      if `true` is passed, points are generated in a random
+ *      and uniform way on the surface of the soup.
+ *      The number of sample points is the value passed to the named
+ *      parameter `number_of_points_on_faces()`. If not set,
+ *      the value passed to the named parameter `number_of_points_per_area_unit()`
+ *      is multiplied by the area of the soup to get the number of sample points.
+ *      If none of these parameters is set, the number of points sampled is `points.size()`.
+ *
+ *      Default is `true`.
+ *    \cgalParamEnd
+ *    \cgalParamBegin{use_grid_sampling}
+ *      if `true` is passed, points are generated on a grid in each triangle,
+ *      with a minimum of one point per triangle. The distance between
+ *      two consecutive points in the grid is that of the length of the
+ *      smallest non-null edge of the soup, or the value passed to
+ *      the named parameter `grid_spacing()`.
+ * 
+ *      Default is `false`.
+ *    \cgalParamEnd
+ *    \cgalParamBegin{use_monte_carlo_sampling}
+ *      if `true` is passed, points are generated randomly in each triangle.
+ *      The number of points per triangle is the value passed to the named
+ *      parameter `number_of_points_per_face()`. If not set, the value passed
+ *      to the named parameter `number_of_points_per_area_unit()` is
+ *      used to pick a number of points per face proportional to the triangle
+ *      area with a minimum of one point per face. If none of these parameters
+ *      is set, 2 divided by the square of the length of the smallest non-null
+ *      edge of the soup is used as if it was passed to
+ *      `number_of_points_per_area_unit()`.
```

Comment:
Too obfuscated, just say: "the number of points per area unit is set to 2/sqrt()..."